### PR TITLE
Drop couch locations

### PR DESCRIPTION
**Pull request by esoergel**
Couch `Location` is dead, long live `SQLLocation`.

I recently removed the last references to couch locations outside of the locations module.  This PR drops the sync code and the code testing it.

I'll later write a mgmt command to delete the documents from couch, but I figure there's no rush on that, and it'd be better to be cautious.

### CODE REVIEW

**Comment by benrudolph** [comment]
holy 💩

File: corehq/apps/dump_reload/couch/dump.py [hunk 1]
```diff
@@ -14,7 +14,6 @@
 from dimagi.utils.couch.database import iter_docs
 
 DOC_PROVIDERS = {
-    DocTypeIDProvider(['Location']),
```
Comment (esoergel) [inline corehq/apps/dump_reload/couch/dump.py]: @snopoke as best as I could tell, this is used to determine what data to save to disk when a particular management command is run.  My assumption is that it's safe to just drop this entirely given that `Location` is being killed, and `SQLLocation` is being dumped anyways, does that sound correct?

**Comment by snopoke** [comment]
Wow, this is awesome!

**Comment by esoergel** [comment]
I think this is good to go.  @snopoke @gcapalbo got any reservations about removing couch locations at this point?  UCR location access has been migrated off couch, as has all code which uses the python `Location` model.  `copy_domain` and similar features also copy SQL locations, so dropping couch locations shouldn't be an issue there.  Are there any places I'm missing which may be accessing the documents in couch directly without going through the model?

**Comment by snopoke** [comment]
Looks good to me. I was wondering about publishing to kafka but looks like that's already set to go.

**Review by snopoke — APPROVED** [review]

**Review by gcapalbo — APPROVED** [review]
@esoergel looks good to me, thanks for doing this.

confirming that the pillow code is using SQLLocations now as well?

**Comment by esoergel** [comment]
@gcapalbo Yup, this gets published to kafka via the SQLLocation, so the UCR pillow reads from that.

**Pull request merged**